src/RenderMethods/RenderFieldTemplate.tsx

```diff
@@ -0,0 +1,132 @@
+import { getCurrentField, getCurrentSection, makeid } from '../utils';
+import { stringDefinition, sectionDefinition } from '../schemaComponents/templates';
+import _ from 'lodash';
+import { JSONSchema7 } from 'json-schema';
+
+// This component is used to render every widget
+// Its purpose here is to add functionality common to every of those such as add or remove
+// It will be used to add drag and drop area later
+
+export default function RenderFieldTemplate({ id, schema, children }: FieldTemplateProps) {
+  const { schema: vernaSchema, setSchema, uiSchema, setUiSchema } = useVerna();
+  const canAddField = id.split('_').length === 3;
+  const canAddSection = id.split('_').length === 2;
+  const isRoot = id === 'root';
+  const ownProperties = Object.keys(schema.properties || {}).length === 0;
+  const currentSection = getCurrentSection(id);
+  const currentField = getCurrentField(id);
+
+  function addItemToSchema(newKey: string) {
+    const newSchema = { ...vernaSchema };
+    if (newSchema.properties) {
+      const section = newSchema.properties[currentSection] as JSONSchema7;
+      if (section.properties) {
+        section.properties[newKey] = stringDefinition(newKey);
+      }
+    }
+    setSchema(newSchema);
+  }
+
+  function addItemToUiSchemaOrder(newKey: string) {
+    const newUiSchema = { ...uiSchema };
+    if (currentSection in newUiSchema && 'ui:order' in newUiSchema[currentSection]) {
+      const currentSectionIndex = _.findIndex(
+        uiSchema[currentSection]['ui:order'],
+        (key) => key === currentField,
+      );
+      newUiSchema[currentSection]['ui:order'].splice(currentSectionIndex + 1, 0, newKey);
+    } else {
+      newUiSchema[currentSection] = {};
+      newUiSchema[currentSection]['ui:order'] = [newKey];
+    }
+    setUiSchema(newUiSchema);
+  }
+
+  function addItem() {
+    const newKey = makeid(10);
+    addItemToSchema(newKey);
+    addItemToUiSchemaOrder(newKey);
+  }
+
+  function addSection() {
+    const newSchema = { ...vernaSchema };
+    const newKey = makeid(10);
+    if (newSchema.properties) {
+      newSchema.properties[newKey] = sectionDefinition(newKey);
+    }
+    setSchema(newSchema);
+    const currentSection = getCurrentSection(id);
+    const currentSectionIndex = _.findIndex(uiSchema['ui:order'], (key) => key === currentSection);
+    const newUiSchema = { ...uiSchema };
+    newUiSchema['ui:order']?.splice(currentSectionIndex + 1, 0, newKey);
+    setUiSchema(newUiSchema);
+  }
+
+  function removeItem() {
+    const newSchema = { ...vernaSchema };
+    if (newSchema.properties) {
+      const sectionProperties = newSchema.properties[currentSection] as JSONSchema7;
+      if (sectionProperties.properties) {
+        delete sectionProperties.properties[currentField];
+      }
+    }
+    setSchema(newSchema);
+    const newUiSchema = { ...uiSchema };
+    newUiSchema[currentSection]['ui:order'] = newUiSchema[currentSection]['ui:order'].filter(
```

Comment:
Sure that `newUiSchema[currentSection]['ui:order']` is always defined?

Reply:
I can add a check, except if the user define a wrong uiSchema it will always be defined.